src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -99,315 +94,195 @@ public void Add(string propertyName, JsonNode propertyValue)
-        /// </exception>
-        /// <exception cref="ArgumentNullException">
-        ///   Provided property name is null.
+        ///   Property name to add already exists.
         /// </exception>
-        public void Add(string propertyName, bool propertyValue) => Add(propertyName, new JsonBoolean(propertyValue));
+        public void Add(string propertyName, JsonArray propertyValue) => Add(propertyName, (JsonNode)propertyValue);
 
         /// <summary>
-        ///   Adds the specified property as a <see cref="JsonNumber"/> to the JSON object.
+        ///   Adds the properties from the specified collection to the JSON object.
         /// </summary>
-        /// <param name="propertyName">Name of the property to add.</param>
-        /// <param name="propertyValue"><see cref="byte"/> value of the property to add.</param>
+        /// <param name="jsonProperties">Properties to add.</param>
         /// <exception cref="ArgumentException">
-        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        ///   Provided collection contains duplicates.
         /// </exception>
         /// <exception cref="ArgumentNullException">
-        ///   Provided property name is null.
+        ///   Some of property names are null.
         /// </exception>
-        public void Add(string propertyName, byte propertyValue) => Add(propertyName, new JsonNumber(propertyValue));
+        public void AddRange(IEnumerable<KeyValuePair<string, JsonNode>> jsonProperties)
+        {
+            foreach (KeyValuePair<string, JsonNode> property in jsonProperties)
+            {
+                Add(property);
+            }
+        }
 
         /// <summary>
-        ///   Adds the specified property as a <see cref="JsonNumber"/> to the JSON object.
+        ///   Removes the property with the specified name.
         /// </summary>
-        /// <param name="propertyName">Name of the property to add.</param>
-        /// <param name="propertyValue"><see cref="short"/> value of the property to add.</param>
-        /// <exception cref="ArgumentException">
-        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
-        /// </exception>
+        /// <param name="propertyName">>Name of a property to remove.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the property is successfully found in a JSON object and removed,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
         /// <exception cref="ArgumentNullException">
         ///   Provided property name is null.
         /// </exception>
-        public void Add(string propertyName, short propertyValue) => Add(propertyName, new JsonNumber(propertyValue));
+        public bool Remove(string propertyName)
```

Comment:
I believe `Remove(string propertyName)` is faster. The overload taking `StringComparison` always iterates through the list until it finds the match, while regular remove should be approaching O(1).  So `Remove(string propertyName, StringComparison.Ordinal)` will do the same thing, but slower.

Reply:
Right, exactly.

So, if someone calls `Remove(string propertyName, StringComparison.Ordinal)`, we should internally special case that enum value and call `Remove(string propertyName)`, since it is equivalent and faster.

Reply:
Right, sure, I'll change it.